tests/integration-tests/tests/common/utils.py

```diff
@@ -90,11 +90,56 @@ def retrieve_latest_ami(region, os, ami_type="official", architecture="x86_64"):
         sys.exit(1)
 
 
+def retrieve_pcluster_ami_without_standard_naming(region, os, version="2.8.1", architecture="x86_64"):
+    try:
+        client = boto3.client("ec2", region_name=region)
+        ami_name = f"pcluster-without-standard-naming-{version}-{os}"
+        amis = client.describe_images(
+            Filters=[{"Name": "name", "Values": [ami_name]}, {"Name": "architecture", "Values": [architecture]}],
+        ).get("Images", [])
+        if amis:
+            # If the AMI is found, return the AMI
+            return amis[0]["ImageId"]
+        else:
+            # If no AMI is found, copy from an official AMI
+            official_ami_name = "aws-parallelcluster-{version}-{ami_name}".format(
+                version=version, ami_name=OS_TO_PCLUSTER_AMI_NAME_OWNER_MAP.get(os).get("name")
+            )
+            official_amis = (
+                boto3.client("ec2", region_name=region)
```

Comment:
why do we call boto3 twice?

Reply:
Thanks, I simplified the code. But I am still keeping the two boto3 calls for simplicity. If we want to use only one boto3 call, we need to use a [paginator](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html) to ensure the first call gets everything we need. However, using paginator leads to possible extra calls